### PR TITLE
HTTPS Client: Add third-party library http-parser from nodejs.

### PR DESCRIPTION
**Pull request by sarenameas**
Add third-party library http-parser from nodejs.
https://github.com/nodejs/http-parser

This PR adds third party library http-parser from nodejs. This is not for comments. This just to show the library going in.

Checklist:
----------

- [n/a] I have tested my changes. No regression in existing tests.
- [n/a] My code is Linted.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: libraries/3rdparty/CMakeLists.txt [hunk 1]
```diff
@@ -1,6 +1,20 @@
 # -------------------------------------------------------------------------------------------------
 # 3rdparty libraries
 # -------------------------------------------------------------------------------------------------
+# http-parser - HTTP message parser
+if(EXISTS "${AFR_3RDPARTY_DIR}/jsmn")
```
Comment (jainamt) [inline libraries/3rdparty/CMakeLists.txt]: ```jsmn``` is not the right thing to check here. This should be ```http-parser```

Comment (sarenameas) [reply libraries/3rdparty/CMakeLists.txt]: Thanks!

**Comment by jainamt** [comment]
Just a question. Do we know which version of this library is being brought over? Is there a tag associated to this version ?

**Comment by sarenameas** [comment]
> Just a question. Do we know which version of this library is being brought over? Is there a tag associated to this version ?

Yes this is version tagged v2.9.0

**Comment by sarenameas** [comment]
I will amend the commit to note so.

**Comment by sarenameas** [comment]
Please note the library lists the version in their header file, http_parser.h:
```
/* Also update SONAME in the Makefile whenever you change these. */
#define HTTP_PARSER_VERSION_MAJOR 2
#define HTTP_PARSER_VERSION_MINOR 9
#define HTTP_PARSER_VERSION_PATCH 0
```

**Pull request merged**